pyanaconda/modules/subscription/runtime.py

```diff
@@ -360,3 +368,200 @@ def run(self):
             # is missing
             message = exception_dict.get("message", _("Failed to attach subscription."))
             raise SubscriptionError(message) from None
+
+
+class ParseAttachedSubscriptionsTask(Task):
+    """Parse data about subscriptions attached to the installation environment."""
+
+    def __init__(self, rhsm_entitlement_proxy, rhsm_syspurpose_proxy):
+        """Create a new attached subscriptions parsing task.
+
+        :param rhsm_entitlement_proxy: DBus proxy for the RHSM Entitlement object
+        :param rhsm_syspurpose_proxy: DBus proxy for the RHSM Syspurpose object
+        """
+        super().__init__()
+        self._rhsm_entitlement_proxy = rhsm_entitlement_proxy
+        self._rhsm_syspurpose_proxy = rhsm_syspurpose_proxy
+
+    @property
+    def name(self):
+        return "Parse attached subscription data"
+
+    @classmethod
+    def _pretty_date(self, date_from_json):
```

Comment:
This should be ``@staticmethod`` and the `self` should be removed.

Reply:
Yep, that's a good idea. :)